core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -351,7 +351,16 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
   def getAppId: String = get("spark.app.id")
 
   /** Does the configuration contain a given parameter? */
-  def contains(key: String): Boolean = settings.containsKey(key)
+  def contains(key: String): Boolean = {
+    if (settings.containsKey(key)) {
+      true
+    } else {
+      // try to find the settings in the alternatives
+      configsWithAlternatives.get(key).flatMap { alts =>
+        alts.collectFirst { case alt if contains(alt.key) => true }
+      }.isDefined
```

Comment:
1. It will find the alternatives using the key: configsWithAlternatives.get(key), it will return an Option of Seq, since alternative could be multiple;
2. It will then try to use the key of each alternative to see if it is already defined in the conf, here contains(alt.key) is a recursive call;
3. collectFirst will just get the first matching
4. the final step isDefined() is to see whether we find such a match or not. 
Remember configsWithAlternatives.get("one") returns Option of Seq[], any suggestion to simplify?